### PR TITLE
Add ZFS Resource Agent

### PR DESCRIPTION
**Pull request by utopiabound**
Adds a resource agent to import/export ZFS pools.

This resource agent has a long history of use and is well tested.

### CODE REVIEW

**Comment by krig** [comment]
Looks generally clean and good to me, but I'm curious why it has a separate debug log mechanism? There is the `ocf_log debug` command built into the OCF library, and the pacemaker `trace_ra` mechanism to get even more detail.

File: heartbeat/ZFS [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+</longdesc>
+<shortdesc lang="en">Import arguments</shortdesc>
+<content type="string" default="" />
+</parameter>
+<parameter name="importforce" unique="1" required="0">
+<longdesc lang="en">
+zpool import is given the -f option.
+</longdesc>
+<shortdesc lang="en">Import is forced</shortdesc>
+<content type="boolean" default="true" />
```
Comment (oalbrigt) [inline heartbeat/ZFS]: You should define and assign "_default" parameters to use here and assign if not set by user (grep for _default in the other agents for examples).

File: heartbeat/ZFS [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        debug_log "starting import"
+
+        # The meanings of the options to import are as follows:
+        #   -f : import even if the pool is marked as imported to another
+        #        system - the system may have failed and not exported it
+        #        cleanly.
+        #   -o cachefile=none : the import should be temporary, so do not
+        #        cache it persistently (across machine reboots). We want
+        #        the CRM to explicitly control imports of this pool.
+	if [ ${OCF_RESKEY_importforce} = 1 ]; then
```
Comment (oalbrigt) [inline heartbeat/ZFS]: ocf_is_true would be better here, as that matches true/on/1 and so on.

**Comment by oalbrigt** [comment]
Looks good to me.

**Pull request merged**